### PR TITLE
add mutex syncronization around Print

### LINKED ISSUE #45: Wrap a mutex around the body of `Print()`.
It's annoying seeing interleave output in concurrent programs and `Print()` has essentially become a debug tool.

### PR DESCRIPTION
**Pull request by ilmanzo**
fix #45

### CODE REVIEW

**Comment by ilmanzo** [comment]
image

CI fails the upload step , do I need to do something on my branch ?

File: printer.go [hunk 1]
```diff
@@ -145,9 +146,12 @@ func Format(v interface{}) string {
 }
 
 var newLine = []byte{'\n'}
+var mux = &sync.Mutex{}
```
Comment (jmalloc) [inline printer.go:149]: This can just be a simple variable declaration, no need for the pointer.

```suggestion
var mux sync.Mutex
```

**Comment by jmalloc** [comment]
Thanks for the PR!

> CI fails the upload step , do I need to do something on my branch ?

No, sorry - that's our CI setup. The secret tokens aren't available in pull-requests from forks. I'll make an issue to sort that out myself.  I can see that the tests are passing, though :)

**Comment by jmalloc** [comment]
Thank you!

**Pull request merged**